packages/Caching/Detector/ChangedFilesDetector.php

```diff
@@ -45,12 +45,6 @@ public function hasFileChanged(SmartFileInfo $smartFileInfo): bool
         return $currentFileHash !== $cachedValue;
     }
 
-    public function invalidateFile(SmartFileInfo $smartFileInfo): void
-    {
-        $fileInfoCacheKey = $this->getFileInfoCacheKey($smartFileInfo);
-        $this->cache->clean($fileInfoCacheKey);
-    }
-
```

Comment:
How is the file invalidated now?

Reply:
It is not invlaidated, inspired by phpstan, when file cache is changed, we analyze that file and then we refresh cache for that file. It works, tested on large code base.